core/store/src/trie/trie_storage.rs

```diff
@@ -144,6 +144,7 @@ impl TrieCacheInner {
                     }
                     None => {
                         self.metrics.shard_cache_pop_misses.inc();
+                        continue;
```

Comment:
Good catch! If we already caught a bug here, let's try to compress this experssion to make it simpler and less error prone. Something like this should work I think: 

```rust
            // First, try to evict value using the key from deletions queue.
            if let Some(key) = self.deletions.pop() {
                match self.cache.pop(&key) {
                    Some(value) => {
                        self.total_size -= value.len() as u64;
                        self.metrics.shard_cache_pop_hits.inc();
                    }
                    None => self.metrics.shard_cache_pop_misses.inc(),
                }
                continue;
            }

```

Reply:
+1 for adding the test (as we shouldn't be in a hurry now).